core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelSet.java

```diff
@@ -83,18 +83,25 @@ DriverChannel next() {
       case 0:
         return null;
       case 1:
-        return snapshot[0];
+        DriverChannel onlyChannel = snapshot[0];
+        return onlyChannel.preAcquireId() ? onlyChannel : null;
       default:
-        DriverChannel best = null;
-        int bestScore = 0;
-        for (DriverChannel channel : snapshot) {
-          int score = channel.getAvailableIds();
-          if (score > bestScore) {
-            bestScore = score;
-            best = channel;
+        while (true) {
+          DriverChannel best = null;
+          int bestScore = 0;
+          for (DriverChannel channel : snapshot) {
+            int score = channel.getAvailableIds();
+            if (score > bestScore) {
```

Comment:
why we are assuming that a higher score means that the channel is better?

Reply:
We want to distribute incoming requests evenly across channels. Even though this is not perfect (some requests might be more expensive than others), I think it's a fair approximation.

Reply:
so it basically means that we are trying to always pick the highest available stream id, right?

Reply:
The channel with the most available ids.